### PR TITLE
Unpin cairo-vm version for cairo1-run crate

### PR DESCRIPTION
**Pull request by fmoletta**
# Unpin cairo-vm version for cairo1-run crate

## Description

Description of the pull request changes and motivation.

## Checklist
- [ ] Linked to Github Issue
- [ ] Unit tests added
- [ ] Integration tests added.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.
  - [ ] CHANGELOG has been updated.

### CODE REVIEW

**Review by juanbono — APPROVED** [review]

**Review by pefontana — APPROVED** [review]

File: cairo1-run/src/main.rs [hunk 1]
```diff
@@ -358,6 +365,203 @@ fn main() -> Result<(), Error> {
...
+    }
+    assert_eq!(before_final_call + final_call_size, ctx.current_code_offset);
+    Ok((ctx.instructions, builtins))
+}
+
+fn get_info<'a>(
+    sierra_program_registry: &'a ProgramRegistry<CoreType, CoreLibfunc>,
+    ty: &'a cairo_lang_sierra::ids::ConcreteTypeId,
+) -> &'a cairo_lang_sierra::extensions::types::TypeInfo {
+    sierra_program_registry.get_type(ty).unwrap().info()
```
Comment (pefontana) [inline cairo1-run/src/main.rs:540]: can we handle this error?

File: cairo1-run/src/main.rs [hunk 2]
```diff
@@ -358,6 +365,203 @@ fn main() -> Result<(), Error> {
...
+        } else if generic_ty == &GasBuiltinType::ID {
+            casm_extend! {ctx,
+                [ap + 0] = initial_gas, ap++;
+            }
+        } else if generic_ty == &SegmentArenaType::ID {
+            let offset = -ap_offset + after_vecs_offset;
+            casm_extend! {ctx,
+                [ap + 0] = [ap + offset] + 3, ap++;
+            }
+            // } else if let Some(Arg::Array(_)) = arg_iter.peek() {
```
Comment (pefontana) [inline cairo1-run/src/main.rs:496]: Are we leaving this commented for a reason?

**Review by pefontana — CHANGES_REQUESTED** [review]
Nice one @fmoletta !
I just leave some comments and we will be good to go!

File: cairo1-run/src/main.rs (hunk 2 above)
Comment (fmoletta) [reply cairo1-run/src/main.rs:496]: Its the part that handles function arguments (which we are not using), I left it for future reference

Comment (pefontana) [reply cairo1-run/src/main.rs:484]: great!

**Review by pefontana — APPROVED** [review]

**Pull request merged**